iron-swipeable-container.html

```diff
@@ -0,0 +1,266 @@
+      } else if (track.state === 'end') {
+        this._trackEnd(track, target);
+      }
+    },
+
+    _trackStart: function(event, target) {
+      // Save the width of the element, so that we don't trigger a style
+      // recalc every time we need it.
+      this._nodeWidth = target.offsetWidth;
+      target.style.transition = 'none';
+    },
+
+    _trackMove: function(event, target) {
+      this._animate(event.dx, target);
+    },
+
+    _trackEnd: function(event, target) {
+      // The element is swiped away if it's moved halfway its total width
+      this._swipeEnd(Math.abs(event.dx) > this._nodeWidth / 2, event.dx > 0, target);
+    },
+
+    _animate: function(x, target) {
+      var direction = x > 0 ? 1 : -1;
+
+      // This is the total distance the animation will take place over.
+      var totalDistance = this._nodeWidth * this.widthRatio;
+
+      // Opacity distance overflow. `this._nodeWidth * this.opacityRate` is the
+      // total distance the element needs to travel to become completely
+      // transparent, and `x` is how much the element has already travelled.
+      var opaqueDistance = Math.max(0, Math.abs(x) - this._nodeWidth * this.opacityRate);
+      var opacity = Math.max(0, (totalDistance - opaqueDistance) / totalDistance);
+      target.style.opacity = opacity
+
+      var translate, rotate;
+
+      if (this.swipeStyle === 'horizontal') {
+        translate = 'translate3d(' + x + 'px,' + 0 + 'px,0)';
+        rotate = '';
+      } else {  // Default is assumed to be `curve`.
+        // Assume the element will be completely transparent at 90 degrees, so
+        // figure out the rotation and vertical displacement needed to
+        // achieve that.
+        var y = totalDistance - Math.sqrt(totalDistance * totalDistance - opaqueDistance * opaqueDistance);
+        var deg = (1 - opacity) * direction * 90;
+
+        translate = 'translate3d(' + x + 'px,' + y + 'px,0)';
+        rotate = ' rotate(' + deg + 'deg)';
+      }
+
+      this.transform(translate + rotate, target);
+    },
+
+    _swipeEnd: function(away, dir, target) {
+      // Restore the original transition;
+      target.style.transition = this.transition;
+      this.away = away;
+      this.direction = dir ? 'right' : 'left';
+
+      if (away) {
+        // If the element is ready to be swiped away, then translate it to the full
+        // transparency distance.
+        var totalDistance = this._nodeWidth * this.widthRatio;
+        this._animate(dir ? totalDistance : -totalDistance, target);
+      } else {
+        this._animate(0, target);
+      }
+    },
+
+    _onTransitionEnd: function(event) {
+      var target = event.currentTarget;
+
+      if (this.away && event.propertyName === 'opacity') {
+        this.fire('iron-swipe-away', {direction: this.direction, target:target});
+      }
+      event.stopPropagation();
+    },
+
+    _swipeAway: function(event) {
+      Polymer.dom(this).removeChild(event.detail.target);
```

Comment:
It seems like it would be better to spare adding a listener for this event, and instead just move this right after 257.